### PR TITLE
AST: Remove LiteralExpr::shallowClone()

### PR DESCRIPTION
**Pull request by slavapestov**
At one point it was used to work around problems with re-type checking
literals in CSApply, but all of our tests appear to pass with this removed.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Review by pschuh — APPROVED** [review]

File: lib/Sema/CSApply.cpp [hunk 1]
```diff
@@ -6685,26 +6685,9 @@ Expr *ExprRewriter::convertLiteralInPlace(Expr *literal,
                                           DeclName builtinLiteralFuncName,
                                           Diag<> brokenProtocolDiag,
                                           Diag<> brokenBuiltinProtocolDiag) {
-  auto &tc = cs.getTypeChecker();
```
Comment (pschuh) [inline lib/Sema/CSApply.cpp]: Too much deleting here.

Comment (slavapestov) [reply lib/Sema/CSApply.cpp]: Yep :)

Comment (slavapestov) [reply lib/Sema/CSApply.cpp]: ... or not enough deleting further down! See updated version :)

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Review by xedin — APPROVED** [review]
Seems reasonable to me.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: test/Constraints/add_with_nil.swift [hunk 2]
```diff
@@ -2,5 +2,5 @@
 
 func test(_ x: Int) -> Int {
   return x + nil
-  // expected-error@-1 {{cannot convert value of type 'Int' to expected argument type '_.Stride'}}
+  // expected-error@-1 {{type of expression is ambiguous without more context}}
```
Comment (slavapestov) [inline test/Constraints/add_with_nil.swift]: @xedin I noticed this diagnostic did change. Are you OK with this? The test case was originally added when a crash was fixed. I don't think the old diagnostic actually gave you any useful information (the real problem is there's no suitable overload of `+` -- as far as I can see the appearance of Stride in the old message is an emergent behavior of CSDiag and CalleeCandidateDiagnostics)

**Pull request merged**